File: ruby/bot.rb
require 'discordrb'
require "discordrb/webhooks"
auth = 'BOT-TOKEN-HERE'
bot = Discordrb::Bot.new token: auth
myid = 666317117154525185
lgsid = 575500747014144000
bot.ready do
bot.update_status("idle", nil, nil)
end
bot.message(with_text: 'ping--') do |event|
m = event.respond('ping?')
t = Time.now - event.timestamp
t = t*1000
m.edit "ping? #{t}ms"
m.delete
curbot = bot.user(bot.profile.id)
h = bot.user(myid)
event.channel.send_embed do |embed|
embed.title = 'Pong'
embed.description = '*ruby*'
embed.fields = [Discordrb::Webhooks::EmbedField.new(name: 'Message Edit Latency',value: "#{t}ms",inline: false)]
embed.footer = Discordrb::Webhooks::EmbedFooter.new(icon_url: "#{h.avatar_url}",text: "created by @#{h.username}##{h.discrim} <#{myid}>")
embed.color = "#592630"
embed.thumbnail = Discordrb::Webhooks::EmbedThumbnail.new(url: curbot.avatar_url)
end
end
bot.message(with_text: 'stopruby--') do |event|
event.respond('i want to kill me too, dw :thumbsup:')
exit
end
bot.message(with_text: 'kys--') do |event|
event.respond('Bye! ***-ruby***')
exit
end
bot.message do |event|
if event.message.author.id != myid and event.message.author.id != lgsid
if event.message.content.start_with?('hook--')
event.respond "no"
end
end
if event.message.author.id == myid or event.message.author.id == lgsid
msg = event.message.content
if msg.start_with?('hook--')
curwebhook = Discordrb::API::Channel.create_webhook "Bot #{auth}",event.message.channel.id,'Alphabet Webhook(bot.rb)'
msgtosend = msg[6..msg.index('|')-1]
event.message.delete
hookid = msg[msg.index('|')+1..msg.length]
hookid = JSON.parse(Discordrb::API::User.resolve "Bot #{auth}",Integer(hookid))
hookid = Discordrb::User.new hookid,bot
curwebhookobj = Discordrb::Webhook.new(JSON.parse(curwebhook), bot)
curwebhookinit = Discordrb::Webhooks::Client.new(token: curwebhookobj.token,id: curwebhookobj.id)
curwebhookinit.execute do |builder|
builder.content = msgtosend
builder.username = hookid.username
builder.avatar_url = hookid.avatar_url
end
Discordrb::API::Webhook.delete_webhook "Bot #{auth}",curwebhookobj.id
end
end
end
bot.run

File: lua/bot.lua
local discordia = require("discordia")
local client = discordia.Client({logLevel = 3})
local enums = discordia.enums
local myid = "666317117154525185"
client:on(
    "ready",
    function()
        print("We have logged in as " .. client.user.tag)
        client:setStatus("idle")
    end
)
function round(x)
    return x + 0.5 - (x + 0.5) % 1
end
client:on(
    "messageCreate",
    function(message)
      if message.author.id ~= client.user.id then
        if message.content == "ping--" then
            local watch = discordia.Stopwatch()
            local msg = message.channel:send("Pong!")
            watch:start()
            msg:setContent("Pong")
            watch:stop()
            local lat = round(watch.milliseconds) .. "ms"
            msg:delete()
            local h = client:getUser(myid)
            message.channel:send {
                embed = {
                    title = "Pong",
                    thumbnail = {url = client.user.avatarURL},
                    description = "*lua*",
                    fields = {
                        {name = "Message Edit Latency", value = lat, inline = false}
                    },
                    color = discordia.Color.fromRGB(89, 38, 48).value,
                    footer = {
                        text = "created by @" .. h.tag .. " <" .. myid .. ">",
                        icon_url = h.avatarURL
                    }
                }
            }
        end
        if message.content == "stoplua--" then
            message.channel:send("Bye! :smile:")
            client:stop()
        end
        if message.content == "kys--" then
            message.channel:send("Bye! ***-lua***")
            client:stop()
        end
    end
  end
)

client:run("Bot BOT-TOKEN-HERE")
